nixos/modules/services/networking/thelounge.nix

```diff
@@ -0,0 +1,79 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.thelounge;
+  dataDir = "/var/lib/thelounge";
+  configJsData = "module.exports = " + builtins.toJSON (
+    { private = cfg.private; port = cfg.port; } // cfg.extraConfig
+  );
+
+  thelounge-wrapper-script = pkgs.writeShellScriptBin "thelounge" ''
+    systemd-run --pty --property=DynamicUser=yes \
```

Comment:
Would a systemd user service and why should it write to `/var/lib/thelounge` instead of the users home?

Reply:
Actually this is a bit out-of-scope of nixos and could be part of home-manager instead.

Reply:
now I got the idea, this is basically used to add new users. But would that actually run the same dynamic user that thelounge was started with?

Reply:
Yup. User-management is done with `thelounge add` etc. This becomes really awkward with systemd's DynamicUser because sessions using `sudo -u thelounge` somehow can't access `/var/lib/thelounge`.

Not sure whether `systemd-run` is the preferred solution. It doesn't seem like there's any other package in the same situation.

Reply:
This will not work. There is a new uid on every run.

```console
$ sudo systemd-run --pty --property=DynamicUser=yes id
Running as unit: run-u1538.service
Press ^] three times within 1s to disconnect TTY.
uid=62575 gid=62575 groups=62575
$ sudo systemd-run --pty --property=DynamicUser=yes id
Running as unit: run-u1537.service
Press ^] three times within 1s to disconnect TTY.
uid=61729 gid=61729 groups=61729
```

I would suggest to create a user using `users.users` without allocating a static uid. When the module is configured for the first time, this will assign a uid and a passwd entry so that sudo can be used.

Reply:
You're missing `--property=StateDirectory=thelounge`.
```
$ sudo systemd-run --pty --property=DynamicUser=yes --property=StateDirectory=thelounge id
Running as unit: run-u751.service
Press ^] three times within 1s to disconnect TTY.
uid=64374 gid=64374 groups=64374
$ sudo systemd-run --pty --property=DynamicUser=yes --property=StateDirectory=thelounge id
Running as unit: run-u755.service
Press ^] three times within 1s to disconnect TTY.
uid=64374 gid=64374 groups=64374
```

Reply:
I don't think this is a good solution. `DynamicUser` works well when the service can run truly isolated, but in this case the user needs to interact with it, so it's not isolated anymore, and this is a workaround for that.

Instead you can just define a new user for thelounge (no static uid needed though) and use that instead of DynamicUser.